### ISSUE TITLE
aws_appsync_graphql_api.authentication_type=OPENID_CONNECT is omitted

### ISSUE THREAD

**Issue by deptno** [terraform-providers/terraform-provider-aws#6134]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Description

appsync authentication_type 'OPENID_CONNECT' is omitted

https://github.com/aws/aws-sdk-go/blob/2742e2e94dd2980c04c4fbe6c342f647f8e6d376/service/appsync/api.go#L5854

It is defined in `aws-sdk-go`, but can't not use.

I think here is related code

https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_appsync_graphql_api.go#L30-L32

but I'm not go developer :sob:

### Terraform Version

```bash
Terraform v0.11.8
+ provider.aws v1.39.0
```

### Affected Resource(s)

* aws_appsync

### Terraform Configuration Files

```
resource "aws_appsync_graphql_api" "dev_readish_oidc" {
  authentication_type = "OPENID_CONNECT"
  name = "some name"
}
```

### Expected Behavior

Run

### Actual Behavior

```bash
Error: aws_appsync_graphql_api.xxx: expected authentication_type to be one of [API_KEY AWS_IAM AMAZON_COGNITO_USER_POOLS], got OPENID_CONNECT
```

**Comment by bflad** [comment]
Pull request submitted to support `OPENID_CONNECT` authentication and its required configuration options: #6138

**Comment by bflad** [comment]
This has been released in [version 1.41.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1410-october-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
